TA.NexDome.DeviceInterface/StateMachine/ControllerStateMachine.cs

```diff
@@ -295,6 +295,15 @@ public void SetHomeSensorAzimuth(decimal azimuth)
             TransitionToState(new RequestStatusState(this));
             }
 
+        public void SyncAzimuth(double azimuth)
```

Comment:
I would suggest delegating this action to the `ReadyState` state and not doing it here immediately in the state machine. In other words, the operation would only be valid when the dome is not moving.

The reason I think that would be better is because of how the firmware tracks position during a slew. The "position" can go outside the range you might expect and gets fixed up when the slew completes. I would worry that altering the position while a slew is in progress could have unexpected effects.